cmd/nitro/nitro.go

```diff
@@ -647,6 +647,45 @@ func mainImpl() int {
 			}
 		}
 	}
+
+	// Before starting the node, wait until the transaction that deployed rollup is finalized
+	if nodeConfig.EnsureRollupDeployment &&
+		nodeConfig.Node.ParentChainReader.Enable &&
+		rollupAddrs.DeployedAt > 0 {
+		currentFinalized, err := l1Reader.LatestFinalizedBlockNr(ctx)
```

Comment:
nit: this would look better in a separate function.

Reply:
wrapping this if block in a function gets dicey when handling with sigInt (where we want to return with exit code 0) not sure if having a separate function is worth it